clients/fides-js/src/fides.ts

```diff
@@ -175,6 +172,13 @@ _Fides = {
   saved_consent: {},
   gtm,
   init,
+  prevConfig: undefined,
+  reinitialize() {
+    if (!this.prevConfig || !this.initialized) {
+      throw new Error("Fides must be initialized before reinitializing");
+    }
+    return this.init(this.prevConfig);
```

Comment:
I know this is maybe really Hard™️, but I think we need to ensure that `Fides.reinitialize` cleans up our existing rendered DOM elements automaticaly... 😄 

I wrote some quick Cypress specs to exercise this and toggled `fides_embed` off and on. Since `reinitialize` doesn't wipe previously rendered elements, I could get into a fun state where I had both an embedded and overlay banner running simultaneously:
![image](https://github.com/ethyca/fides/assets/1834295/467b76fe-5dfd-43cf-b22e-68f31884d609)

Reply:
What do you mean, that's a FEATURE!

Reply:
OK, I added some code to `initOverlay` to unmount previously rendered instances, which makes this work swimmingly.